main/SAPI.c

```diff
@@ -1095,6 +1095,54 @@ SAPI_API void sapi_terminate_process(TSRMLS_D) {
 	}
 }
 
+SAPI_API void sapi_add_request_header(char *var, unsigned int var_len, char *val, unsigned int val_len, void *arg TSRMLS_DC) /* {{{ */
+{
+	zval *return_value = (zval*)arg;
+	char *str = NULL;
+	char *p;
+	ALLOCA_FLAG(use_heap)
+
+	if (var_len > 5 &&
+		var[0] == 'H' &&
+		var[1] == 'T' &&
+		var[2] == 'T' &&
+		var[3] == 'P' &&
+		var[4] == '_') {
```

Comment:
I didn't checked for FPM, but doing a "strace" on a FastCGI-only process shows that the headers are transmitted in upper case, no matter what they looked like in the original request. I guess FPM will receive the same data when using mod_fastcgi on Apache.

Might be different with other implementations, like mod_proxy_fcgi (or other web server), but an existing case sensitive code would already not work is most cases.